compiler/semexprs.nim

```diff
@@ -1577,9 +1577,13 @@ proc takeImplicitAddr(c: PContext, n: PNode; isLent: bool): PNode =
         root.name.s, renderTree(n, {renderNoComments}), explanationsBaseUrl])
   case n.kind
   of nkHiddenAddr, nkAddr: return n
-  of nkHiddenDeref, nkDerefExpr: return n[0]
```

Comment:
I don't understand this patch.

Reply:
var params are nkHiddenDeref, and before PR, would loose their nkHiddenDeref, so `result = param` would deref instead of taking by address, causing a bug in cgen (result = param instead of result = &param)

I've now added plenty of tests regarding `var result` and identified 2 pre-existing bugs (not fixed in this PR)